pages/payment.js

```diff
@@ -0,0 +1,281 @@
+import Head from 'next/head';
+import Header from '../components/Header.js';
+import Footer from '../components/Footer.js';
+import Link from 'next/link';
+import cookie from 'js-cookie';
+import React from 'react';
+import nextCookies from 'next-cookies';
+import { useState, useEffect } from 'react';
+// import PaymentInputs from '../components/PaymentInputs.js';
+
+export default function Payment() {
+  const [input, setInput] = useState('');
+
+  // useEffect(() => {
+  //   setInput(input);
+  // }, [input]);
+
+  function buy() {
+    cookie.remove('cart');
+    cookie.remove('total');
+  }
+
+  const regexText = /[a-zA-Z]/g;
+  const regexNumber = /[0-9/]/g;
+
+  function checksTypeText(e) {
+    if (e.target.value.match(regexText)) {
+      setInput(e.target.value);
+      console.log(input);
+    } else {
+      e.target.value = '';
+    }
+  }
+
+  function checksTypeNumber(e) {
+    if (e.target.value.match(regexNumber)) {
+      setInput(e.target.value);
+      console.log(input);
+    } else {
+      e.target.value = '';
+    }
+  }
```

Comment:
I guess these are for data validation? These regular expression patterns will only check whether there is a letter or number **somewhere** in the text.

So, the following are valid:

- Full name: "Alenio8"
- Zip Code: "1abc"